internal/session.go

```diff
@@ -540,9 +540,20 @@ func (env *sessionEnvironmentImpl) AddSessionToken() {
 
 func (env *sessionEnvironmentImpl) SignalCreationResponse(ctx context.Context, sessionID string) error {
 	activityEnv := getActivityEnv(ctx)
-	client := activityEnv.serviceInvoker.GetClient(activityEnv.workflowDomain, &ClientOptions{})
-	return client.SignalWorkflow(ctx, activityEnv.workflowExecution.ID, activityEnv.workflowExecution.RunID,
-		sessionID, env.getCreationResponse())
+
+	signalInput, err := encodeArg(activityEnv.dataConverter, env.getCreationResponse())
```

Comment:
Just to clarify, the use of `activityEnv.dataConverter` instead of the default data converter is a bug fix?

Reply:
No bug. This is actually a good observation. Signal input here is controller by client itself, so I should probably leave default converter.

Reply:
Updated.